_P176_CDM7160.ino

```diff
@@ -236,74 +171,72 @@ boolean Plugin_176(byte function, struct EventStruct *event, String& string)
       }
   case PLUGIN_INIT:
       {
-      plugin_176_setPowerDown(); // disabled because there were issues setting alt in powerdown mode
-      unsigned char elev = Settings.TaskDevicePluginConfig[event->TaskIndex][1] / 10;
-      delay(100);
-      if (elev)
-      {
-       plugin_176_setAltitude(elev);
-      } else
-      {
-       plugin_176_clearAltitude();
-      }
-      delay(100);
-      plugin_176_setContinuous();
-      success = true;
+        plugin_176_setPowerDown(); // disabled because there were issues setting alt in powerdown mode
+        unsigned char elev = Settings.TaskDevicePluginConfig[event->TaskIndex][1] / 10;
+        delay(100);
+        if (elev)
+        {
+          plugin_176_setAltitude(elev);
+        } else
+        {
+          plugin_176_clearAltitude();
+        }
+        delay(100);
+        plugin_176_setContinuous();
+        success = true;
       }
     case PLUGIN_READ:
       {
         plugin_176_i2caddr = Settings.TaskDevicePluginConfig[event->TaskIndex][0];
 
         plugin_176_begin();
-         unsigned int co2;
-     unsigned char status;
+        unsigned int co2;
+        unsigned char status;
      
-     unsigned char count;
-     for (count = 0; count < 6; count++)
-     {
-      if (plugin_176_getStatus(status))
-      {
-        if (status & CDM7160_FLAG_BUSY)
-        {
-          delay(50); // BUSY may take up to 300ms
-        } else
+        unsigned char count;
+        for (count = 0; count < 6; count++)
         {
-          break;
-        }
-      }
-     }
-     UserVar[event->BaseVarIndex + 1] = count;
-          
-         if (plugin_176_getCO2(co2))
+         if (plugin_176_getStatus(status))
          {
-           boolean good;  // True if sensor is not saturated
-           good = (co2 <= 10000);
-           UserVar[event->BaseVarIndex] = co2;
-           if (!good)
+           if (status & CDM7160_FLAG_BUSY)
+           {
+             delay(50); // BUSY may take up to 300ms
```

Comment:
Sensor measures CO2 concentration every 2 seconds and it takes 0.3s. Value shouldn't be read during this time and condition is signaled by BUSY bit. I assumed other tasks like Wifi may be handled during delay periods. I'll check BME280 code.

Reply:
A blocking call for 300 msec is too long. I guess you can still read the result even when returning to it after more than 300 msec. So that should be split into stages if a delay is needed.
If no delay is needed, you can continue the normal flow, but if it is needed, please split it and do the reading in a call to 10/sec case like in the BME280 code.

Reply:
PLUGIN_ONCE_A_SECOND like in BME280 should be enough.